graylog2-server/src/main/java/org/graylog2/database/dbcatalog/DbEntitiesScanner.java

```diff
@@ -18,71 +18,40 @@
 
 import com.google.inject.Inject;
 import com.google.inject.Provider;
-import org.graylog2.database.DbEntity;
+import org.graylog2.database.dbcatalog.impl.ClassGraphDbEntitiesScanningMethod;
+import org.graylog2.database.dbcatalog.impl.DbEntitiesScanningMethod;
 import org.graylog2.shared.plugins.ChainingClassLoader;
-import org.reflections.Reflections;
-import org.reflections.scanners.Scanners;
-import org.reflections.util.ConfigurationBuilder;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.util.List;
-import java.util.stream.Collectors;
-import java.util.stream.Stream;
-
 public class DbEntitiesScanner implements Provider<DbEntitiesCatalog> {
 
     private static final Logger LOG = LoggerFactory.getLogger(DbEntitiesScanner.class);
 
     private final String[] packagesToScan;
-
-    private final ClassLoader chainingClassLoader;
+    private final String[] packagesToExclude;
+    private final ChainingClassLoader chainingClassLoader;
+    private final DbEntitiesScanningMethod dbEntitiesScanningMethod = new ClassGraphDbEntitiesScanningMethod();
 
     @SuppressWarnings("unused")
     @Inject
     public DbEntitiesScanner(final ChainingClassLoader chainingClassLoader) {
         this.chainingClassLoader = chainingClassLoader;
         this.packagesToScan = new String[]{"org.graylog2", "org.graylog"};
+        this.packagesToExclude = new String[]{"org.graylog.shaded", "org.graylog.storage", "org.graylog2.migrations"};
     }
 
-    DbEntitiesScanner(String[] packagesToScan) {
+    DbEntitiesScanner(final String[] packagesToScan, String[] packagesToExclude) {
```

Comment:
Is there a specific reason to declare `packagesToScan` as `final` but `packagesToExclude` not?

Reply:
Nope, I have just missed that, thx!